### PR TITLE
PXB-3162 Describe the options available for xtrabackup redo logs (8.0)

### PR DESCRIPTION
**Pull request by patrickbirch**
new file:   docs/redo-log-archiving.md
	modified:   mkdocs-base.yml

### CODE REVIEW

File: docs/redo-log-archiving.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# InnoDB redo log archiving
+
+A physical backup contains raw copies of the database contents and any related files, such as configuration files or logs. During crash recovery, InnoDB corrects the data from incomplete transactions using the redo log. A redo log contains redo log files. An LSN value represent the data in the redo log.
```
Comment (alina-derkach-oaza) [inline docs/redo-log-archiving.md]:
```suggestion
A physical backup contains raw copies of the database contents and any related files, such as configuration files or logs. During crash recovery, InnoDB corrects the data from incomplete transactions using the redo log. A redo log contains redo log files. An LSN value represents the data in the redo log. 
```

File: docs/redo-log-archiving.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+# InnoDB redo log archiving
+
+A physical backup contains raw copies of the database contents and any related files, such as configuration files or logs. During crash recovery, InnoDB corrects the data from incomplete transactions using the redo log. A redo log contains redo log files. An LSN value represent the data in the redo log. 
+
+Data in the redo log is appended and the movement of the checkpoint truncates old data. 
+If the MySQL server has significant activity during a backup operation and the redo log file storage media operates faster than the backup storage media, the backup operation may fail to keep pace with the redo log generation and cause a loss of data.
```
Comment (alina-derkach-oaza) [inline docs/redo-log-archiving.md]:
```suggestion
If the MySQL server has significant activity during a backup operation and the redo log file storage media operates faster than the backup storage media, the backup operation may fail to keep pace with the redo log generation and cause data loss.
```

File: docs/redo-log-archiving.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+# InnoDB redo log archiving
+
+A physical backup contains raw copies of the database contents and any related files, such as configuration files or logs. During crash recovery, InnoDB corrects the data from incomplete transactions using the redo log. A redo log contains redo log files. An LSN value represent the data in the redo log. 
+
+Data in the redo log is appended and the movement of the checkpoint truncates old data.
```
Comment (alina-derkach-oaza) [inline docs/redo-log-archiving.md]:
```suggestion
Data in the redo log is appended, and the movement of the checkpoint truncates old data. 
```

File: docs/redo-log-archiving.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+* Checking for I/O congestion if the read speed is slow
+
+* Checking for I/O or network congestion if the write speed is slow
+
+* Taking the backup at an off-peak time
+
+## redo log archiving
+
+MySQL 8.0.17 added redo log archiving. To enable this feature, the follow settings are required:
```
Comment (alina-derkach-oaza) [inline docs/redo-log-archiving.md]:
```suggestion
MySQL 8.0.17 added redo log archiving. To enable this feature, the following settings are required:
```

File: docs/redo-log-archiving.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    |                                                   0 |
+    +------------------- ---------------------------------+
+    1 row in set (0.00225 sec)
+    ```
+
+## Percona XtraBackup and redo log archiving
+
+Percona XtraBackup 8.0.34-29 added the [--redo-log-arch-dir](xtrabackup-option-reference.md#-redo-log-arch-dirname) option. This option lets you configure a redo log archiving directory, if this directory is not already configured in MySQL.
+
+To use this option you must run Percona XtraBackup as the owner of `mysqld`. If Percona XtraBackup does not have the appropriate permissions, then the action generates a "Permission denied" error and archiving is not used. Percona XtraBackup creates a temporary directory for the archive file.
```
Comment (alina-derkach-oaza) [inline docs/redo-log-archiving.md]:
```suggestion
To use this option, you must run Percona XtraBackup as the owner of `mysqld`. If Percona XtraBackup does not have the appropriate permissions, then the action generates a "Permission denied" error, and archiving is not used. Percona XtraBackup creates a temporary directory for the archive file.
```

File: docs/xtrabackup-option-reference.md [hunk 6]
```diff
@@ -735,7 +735,7 @@ option only has effect with the `--prepare` and
 
 ### --redo-log-arch-dir(=name)
 
-This option sets the redo log archive directory if this directory is not already set on the server.
+This option sets the redo log archive directory if this directory is not already set on the server. To use this option you must run Percona XtraBackup as the owner of `mysqld`.
```
Comment (alina-derkach-oaza) [inline docs/xtrabackup-option-reference.md]:
```suggestion
This option sets the redo log archive directory if this directory is not already set on the server. To use this option, you must run Percona XtraBackup as the owner of `mysqld`.
```

**Review by alina-derkach-oaza — APPROVED** [review]

File: docs/xtrabackup-option-reference.md [hunk 7]
```diff
@@ -733,9 +733,9 @@ Uses the given number of threads to rebuild indexes in a compact backup. This
 option only has effect with the `--prepare` and
 `--rebuild-indexes` options.
 
-### --redo-log-arch-dir(=name)
+### redo-log-arch-dir(=name)
 
-This option sets the redo log archive directory if this directory is not already set on the server.
+This option sets the redo log archive directory if this directory is not already set on the server. To use this option, you must run Percona XtraBackup as the owner of `mysqld`.
```
Comment (altmannmarcelo) [inline docs/xtrabackup-option-reference.md:742]: > To use this option, you must run Percona XtraBackup as the owner of `mysqld`.

I would re-word this part. Perhaps use `... Percona XtraBackup as the same OS user as mysqld`

File: docs/redo-log-archiving.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
+# InnoDB redo log archiving
+
+A physical backup contains raw copies of the database contents and any related files, such as configuration files or logs. During crash recovery, InnoDB corrects the data from incomplete transactions using the redo log. A redo log contains redo log files. An LSN value represents the data in the redo log.
```
Comment (altmannmarcelo) [inline docs/redo-log-archiving.md]:
```suggestion
A physical backup contains raw copies of the database contents and any related files, such as configuration files or logs. During crash recovery, InnoDB corrects the data from incomplete transactions using the redo log. A redo log contains entries necessary to bring the database back to a consistent state.
```

File: docs/redo-log-archiving.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+Data in the redo log is appended, and the movement of the checkpoint truncates old data.
```
Comment (altmannmarcelo) [inline docs/redo-log-archiving.md]:
```suggestion
Data in the redo log is always appended. When a file is full, a new file is created and the movement of the checkpoint truncates old files.
```

File: docs/redo-log-archiving.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+    1 row in set (0.00225 sec)
+    ```
+
+## Percona XtraBackup and redo log archiving
+
+Percona XtraBackup 8.0.34-29 added the [redo-log-arch-dir](xtrabackup-option-reference.md#-redo-log-arch-dirname) option. If the archiving directory is not configured in MySQL, Percona XtraBackup creates a temporary directory for the archive file. To use this option, you must run Percona XtraBackup as the owner of `mysqld`. If Percona XtraBackup does not have the appropriate permissions, then the action generates a "Permission denied" error, and archiving is not used.
+
+```{.bash data-prompt}
+$ sudo -H -u mysql bash
+$ xtrabackup --no-lock=1 --compress --parallel=4 --host=localhost --user=root --password='password_string' --backup=1 --target-dir=/Backup/13oct 2> /tmp/b0-with-redo-archiving-as-mysql-os-user.log
```
Comment (altmannmarcelo) [inline docs/redo-log-archiving.md:82]: Where is the `--redo-log-arch-dir` parameter been set?

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]

File: docs/redo-log-archiving.md (hunk 10 above)
Comment (patrickbirch) [reply docs/redo-log-archiving.md:82]: I don't understand the question.

**Pull request merged**